core/app/models/spree/tax/output/item.rb

```diff
@@ -0,0 +1,10 @@
+module Spree
+  module Tax
+    module Output
+      class Item
+        include ActiveModel::Model
+        attr_accessor :id, :tax_rate, :amount, :included_in_price
```

Comment:
I appreciate both sides here. We do want to associate tax amounts with specific items, and an ID is the most obvious and robust way to do that. That said, the more abstracted from the DB this is the better.

It's worth considering that passing using an ID here doesn't necessarily require that the ID is sent over the wire to an external service, the sort-by-id could be done in the extension.